casper/src/main/scala/coop/rchain/casper/Validate.scala

```diff
@@ -31,15 +28,59 @@ object Validate {
       if (isValid) {
         true.pure[F]
       } else {
-        Log[F].warn(ignore(b, "signature is invalid.")) *> false.pure[F]
+        for {
+          _ <- Log[F].warn(ignore(b, "signature is invalid."))
+        } yield false
       }
     } else {
-      Log[F].warn(
-        ignore(b, s"signature algorithm ${b.sigAlgorithm} is unsupported.")
-      ) *> false.pure[F]
+      for {
+        _ <- Log[F].warn(ignore(b, s"signature algorithm ${b.sigAlgorithm} is unsupported."))
+      } yield false
+    }
+
+  def blockSender[F[_]: Applicative: Log](b: BlockMessage,
+                                          genesis: BlockMessage,
+                                          dag: BlockDag): F[Boolean] =
+    if (b == genesis) {
+      true.pure[F] //genesis block has a valid sender
+    } else {
+      val weight = ProtoUtil.weightFromSender(b, dag.blockLookup)
+      if (weight > 0)
+        true.pure[F]
+      else
+        for {
+          _ <- Log[F].warn(
+                ignore(b, s"block creator ${PrettyPrinter.buildString(b.sender)} has 0 weight."))
+        } yield false
     }
 
-  def blockNumber[F[_]: Applicative: Log](b: BlockMessage, dag: BlockDag): F[Boolean] = {
+  def validateBlockSummary[F[_]: Monad: Log](block: BlockMessage,
+                                             genesis: BlockMessage,
+                                             dag: BlockDag): F[Either[BlockStatus, BlockStatus]] =
+    for {
+      status <- Validate.missingBlocks[F](block, dag)
+      status <- status.traverse(_ => Validate.blockNumber[F](block, dag))
```

Comment:
You can reuse the name `status` over and over, and that works? Weird... I don't really like it though and would prefer if each was named differently. E.g. `missingStatus`, `numberStatus`, `parentsStatus`, etc.